crates/uv-resolver/src/lock.rs

```diff
@@ -713,36 +710,45 @@ impl Wheel {
         }
     }
 
-    fn from_dist(dist: &Dist, hashes: &[HashDigest]) -> Result<Option<Wheel>, LockError> {
+    fn from_dist(dist: &Dist, hashes: &[HashDigest]) -> Result<Vec<Wheel>, LockError> {
         match *dist {
-            Dist::Built(ref built_dist) => Wheel::from_built_dist(built_dist, hashes).map(Some),
-            Dist::Source(_) => Ok(None),
+            Dist::Built(ref built_dist) => Wheel::from_built_dist(built_dist, hashes),
+            Dist::Source(_) => Ok(vec![]),
         }
     }
 
-    fn from_built_dist(built_dist: &BuiltDist, hashes: &[HashDigest]) -> Result<Wheel, LockError> {
+    fn from_built_dist(
+        built_dist: &BuiltDist,
+        hashes: &[HashDigest],
+    ) -> Result<Vec<Wheel>, LockError> {
         match *built_dist {
             BuiltDist::Registry(ref reg_dist) => Wheel::from_registry_dist(reg_dist),
             BuiltDist::DirectUrl(ref direct_dist) => {
-                Ok(Wheel::from_direct_dist(direct_dist, hashes))
+                Ok(vec![Wheel::from_direct_dist(direct_dist, hashes)])
             }
-            BuiltDist::Path(ref path_dist) => Ok(Wheel::from_path_dist(path_dist, hashes)),
+            BuiltDist::Path(ref path_dist) => Ok(vec![Wheel::from_path_dist(path_dist, hashes)]),
         }
     }
 
-    fn from_registry_dist(reg_dist: &RegistryBuiltDist) -> Result<Wheel, LockError> {
-        let filename = reg_dist.filename.clone();
-        let url = reg_dist
-            .file
-            .url
-            .to_url()
-            .map_err(LockError::invalid_file_url)?;
-        let hash = reg_dist.file.hashes.first().cloned().map(Hash::from);
-        Ok(Wheel {
-            url,
-            hash,
-            filename,
-        })
+    fn from_registry_dist(reg_dist: &RegistryBuiltDist) -> Result<Vec<Wheel>, LockError> {
+        let mut wheels = vec![];
+        for wheel in &reg_dist.wheels {
+            let filename = wheel.filename.clone();
+            let url = wheel
+                .file
+                .url
+                .to_url()
+                .map_err(LockError::invalid_file_url)?;
+            // FIXME: Is it guaranteed that there is at least one hash?
+            // If not, we probably need to make this fallible.
+            let hash = Some(Hash::from(wheel.file.hashes[0].clone()));
```

Comment:
It looks like this got changed back from `let hash = reg_dist.file.hashes.first().cloned().map(Hash::from);`. Is that intentional?

Reply:
Ah no, not intentional. Maybe a bad merge. Or I fumbled conflict resolution.

Reply:
I switched it back.